### DISCUSSION TITLE
ANT Compiler - Class (.cls) files compiled and stored under Main Directory regardless of subfolder location of original files

### DISCUSSION THREAD

**Discussion by happy-365** [Riverside-Software/pct#528] [Q&A]
Hi all,

The problem I'm experiencing with ANT compiler is that all my compiled .cls files are stored as a subfolder from the main folder regardless of the .cls file's location.

e.g. compiling c:\mainfolder\src\classes will store the resulting .r files  in the c:\mainfolder\classes folder.

What is strange is that it only applies to .cls files. Compiled .p /.w files in c:\mainfolder\classes are saved in the correct folders.
Just only seems to occur with .cls files.

ANT compiler I was using was part of Progress OpenEdge 12.8, and Progress Support redirected me here so I apologise in advance if I am entering the issue in the wrong forum.

**Comment by cverbiest** [comment]
@happy-365
classes are compiled into their package subdirectory.
what is the correct folder and how do you specify it ?
It's probably best to use the bug report issue template .

**Reply by happy-365** [reply]
Regarding the correct folder, I would like it compiled in the folder where the original .cls files lives.
For now, .cls files compiled in subdirectory level 2+, will create the folder and store it in the level 1 subdirectory.

**Reply by gquerret** [reply]
Please include the PCTCompile statement that you're using and a class example.

**Reply by happy-365** [reply]
Hi, I only run one command and we compile everything in the folder.

From command prompt > C:\Progress\OpenEdge12\ant\bin\ant buildAllOE12 -v
Which compiles all files in directory and subdirectory based on build.xml contents.
The main section is below (buildallOE12 calls 'compileoe12'):

  <target name = "compileOE12" description="Builds OE12 source files" >
    <echo message="Compiling encrypted resources" />
    <PCTCompile destDir="./" dlcHome="${env.DLC}" noParse="true" displayFiles="1"
      paramFile="${paramOE12.file}" iniFile="${iniOE12.file}" graphicalMode="true" xcode="true"
      xCodeSessionKey="XXXXXXX">
      <fileset dir="./" includes="p-crypt.p" />
    </PCTCompile>
    <echo message="Compiling other resources" />
    <PCTCompile destDir="./" dlcHome="${env.DLC}" noParse="true" displayFiles="1"
      paramFile="${paramOE12.file}" iniFile="${iniOE12.file}" graphicalMode="true">
      <fileset dir="./" includes="**/*.p **/*.w **/*.cls"
        excludes="procedures/** p-crypt.p" excludesfile="./jenkins/exclude-filesOE12.txt" />
    </PCTCompile>
  </target>

I can't see what setting i'm missing to have it compile in the same directory the .cls lives (if it's on subfolder level greater than 1)

**Reply by gquerret** [reply]
We can't answer without details of your build script. Open a ticket [here](https://github.com/Riverside-Software/pct/issues/new?assignees=&labels=&projects=&template=bug_report.md&title=) and make sure you provide enough info (especially the PCTCompile statement in build.xml and a .cls example).
